File: support/src/main/java/com/bhm/sdk/support/DownloadRequest.java
package com.bhm.sdk.support;

/**
 * @author Buhuiming
 * @description: 下载管理
 * @date :2022/10/19 15:19
 */
public class DownloadRequest implements IRequest{

    @Override
    public void newRequest(DownloadConfig config) {
        DownloadManager.getInstance().init(config);
    }

    @Override
    public boolean startDownload(String url) {
        return false;
    }

    @Override
    public boolean startDownloads(String[] urls) {
        return false;
    }

    @Override
    public boolean startAllDownloads() {
        return false;
    }

    @Override
    public boolean pauseDownload(String url) {
        return false;
    }

    @Override
    public boolean pauseDownloads(String[] urls) {
        return false;
    }

    @Override
    public boolean pauseAllDownloads() {
        return false;
    }

    @Override
    public boolean removeDownload(String url) {
        return false;
    }

    @Override
    public boolean removeDownloads(String[] urls) {
        return false;
    }

    @Override
    public boolean removeAllDownloads() {
        return false;
    }
}

File: support/src/main/java/com/bhm/sdk/support/DownloadRequest.kt
package com.bhm.sdk.support

import android.app.Application
import android.app.Notification
import com.bhm.sdk.support.interfaces.IRequest

/**
 * @author Buhuiming
 * @description: 下载管理
 * @date :2022/10/19 15:19
 */
class DownloadRequest(private val context: Application) : IRequest {

    override fun newRequest(config: DownloadConfig) {
        DownloadManager.getInstance(context)?.newCall(config)
    }

    override fun startDownload(url: String, fileName: String): Boolean? {
        return DownloadManager.getInstance(context)?.startDownload(url, fileName)
    }

    override fun reStartDownload(url: String, fileName: String): Boolean? {
        return DownloadManager.getInstance(context)?.reStartDownload(url, fileName)
    }

    override fun pauseDownload(url: String, fileName: String): Boolean? {
        return DownloadManager.getInstance(context)?.pauseDownload(url, fileName)
    }

    override fun deleteDownload(url: String, fileName: String): Boolean? {
        return DownloadManager.getInstance(context)?.deleteDownload(url, fileName)
    }

    fun updateNotification(notification: Notification?) {
        DownloadManager.getInstance(context)?.updateNotification(notification)
    }
}